app/controllers/api/v1/saved_scenarios_controller.rb

```diff
@@ -5,6 +5,7 @@ class SavedScenariosController < BaseController
 
       before_action :require_user, only: :index
       load_and_authorize_resource(class: SavedScenario, only: %i[index create show update destroy])
+      before_action :load_saved_scenario, only: :show
```

Comment:
```suggestion
      load_and_authorize_resource(class: SavedScenario, only: %i[index create show update destroy])
```
The exact thing you put in the `load_saved_scenario` method is already done by `load_and_authorize_resource`, which means `@saved_scenario` is already available. Sorry for the trick question!